File: library/src/main/java/org/fbme/lib/common/StringIdentifier.java
package org.fbme.lib.common;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class StringIdentifier implements Identifier {

    private final @NotNull String myValue;

    public StringIdentifier(@NotNull String value) {
        myValue = value;
    }

    public @NotNull String getValue() {
        return myValue;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StringIdentifier that = (StringIdentifier) o;
        return Objects.equals(myValue, that.myValue);
    }

    @Override
    public int hashCode() {
        return myValue.hashCode();
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/common/StringIdentifier.kt
package org.fbme.lib.common

class StringIdentifier(val value: String) : Identifier {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as StringIdentifier
        return value == that.value
    }

    override fun hashCode(): Int {
        return value.hashCode()
    }
}
